### ISSUE TITLE
Incorrect keyboard operation of navigation blocks with nested submenus.

### ISSUE THREAD

**Issue by torounit** [WordPress/gutenberg#64626]
### Description

Related: #60738

When using the keyboard to navigate a navigation block with nested submenus,
If you open and close a nested submenu and then try to open the next nested submenu, the submenu will be closed.

This occurs regardless of the “Open on click” setting.

### Step-by-step reproduction instructions

1. Create the following navigation

```
Top Level Menu Item
    1. Submenu item
        1.1. Nested Submenu Item
        1.2. Nested Submenu Item
    2. Submenu item
        2.1. Nested Submenu Item
        2.2. Nested Submenu Item
```

2. Save post and Show the post.
3. Check that the submenus can be navigated using the keyboard.
4. Open and close nested submenus and use tab to move to the next submenu.

### Screenshots, screen recording, code snippet

https://github.com/user-attachments/assets/cd2a616b-0c19-4ea5-93f8-f26bac4df088

The navigation block is coded as follows

```
<!-- wp:navigation {"openSubmenusOnClick":true} -->
	<!-- wp:navigation-submenu {"label":"Top Level Menu Item","url":"#","kind":"custom"} -->
		<!-- wp:navigation-submenu {"label":"1. Submenu Item","url":"#","kind":"custom"} -->
			<!-- wp:navigation-link {"label":"1.1. Nested Submenu Item","url":"#","kind":"custom"} /-->
			<!-- wp:navigation-link {"label":"1.2. Nested Submenu Item","url":"#","kind":"custom"} /-->
			<!-- wp:navigation-link {"label":"1.3. Nested Submenu Item","url":"#","kind":"custom"} /-->	
		<!-- /wp:navigation-submenu -->
		<!-- wp:navigation-submenu {"label":"2. Submenu Item","url":"#","kind":"custom"} -->
			<!-- wp:navigation-link {"label":"2.1. Nested Submenu Item","url":"#","kind":"custom"} /-->
			<!-- wp:navigation-link {"label":"2.2. Nested Submenu Item","url":"#","kind":"custom"} /-->
			<!-- wp:navigation-link {"label":"2.3. Nested Submenu Item","url":"#","kind":"custom"} /-->
		<!-- /wp:navigation-submenu -->
	<!-- /wp:navigation-submenu -->
<!-- /wp:navigation -->
```

### Environment info

- WordPress 6.6.1
- Gutenberg 19
- Chrome 127 on macOS 14.6.1

### Please confirm that you have searched existing issues in the repo.

- [X] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [X] Yes

**Comment by himanshupathak95** [comment]
After testing this navigation menu keyboard interaction issue, I've found that the menu navigation appears to function as expected ( I am running on the latest trunk ). The nested submenus seem to open and close correctly using keyboard controls.

It's possible this issue may have been resolved in recent releases, but we want to ensure comprehensive verification before closing the ticket. So adding a `needs-testing` label on this might aid in resolution.


My testing environment - 

- WordPress: 6.9-alpha-60176
- PHP: 8.2
- Browser: Chrome 
- OS: macOS
- Theme: Twenty Twenty-Five 1.2

**Comment by t-hamano** [comment]
This issue should have been fixed in #66872. See https://github.com/WordPress/gutenberg/issues/60738#issuecomment-2780644404
